### PR TITLE
Refactor plot.py

### LINKED ISSUE #2867: Make the photometry/spectroscopy height responsive
Right now when you want to use all the new buttons below one of these plots on the source page (photometry or spectroscopy) you have to scroll down inside the accordion containing the plot. 

I think it would be much nicer to have the height of the accordion fit all the buttons so at least you wouldn't need to scroll inside it. 

It would make the accordions bigger but since they are foldable that shouldn't make much of a difference to the users.

### PR DESCRIPTION
**Pull request by jadalilleboe**
To aid the spring cleaning effort I've refactored `skyportal/plot.py`. Previously there was some duplicated code with creating the photometry panels, I've extracted a lot of that out to some helper functions. Also closes #2867

### CODE REVIEW

File: skyportal/plot.py [hunk 1]
```diff
@@ -547,57 +567,684 @@ def make_custom_buttons_div(panel_name):
...
+    panel_name : str
+        Name of the panel. One of 'flux', 'mag' or 'period'
+    color_dict : dict
+        Dictionary defining which color the data points should be
+    y_range : tuple(int)
+        Tuple containing the minimum and maximum y range values of the plot respectively
+    data : pandas DataFrame object
+        Photometry data (ungrouped)
+    obsind : pandas Series object
+        ???
```
Comment (jadalilleboe) [inline skyportal/plot.py:811]: If someone could help with the documentation here, that would be great. [code here](https://github.com/skyportal/skyportal/blob/master/skyportal/plot.py#L639)

Comment (mcoughlin) [reply skyportal/plot.py:811]: @jadalilleboe So I see
    obsind = data['hasflux'] & (
        data['flux'].fillna(0.0) / data['fluxerr'] >= PHOT_DETECTION_THRESHOLD
    )
So I would describe this as
   Index of data that exceed the required signal to noise ratio threshold
   
 What do you think @guynir42 ?

Comment (guynir42) [reply skyportal/plot.py:811]: Maybe "index of points that have been positively detected (observed), i.e., points that are not upper limits."

File: skyportal/plot.py [hunk 2]
```diff
@@ -547,57 +567,684 @@ def make_custom_buttons_div(panel_name):
...
+                period_double_button,
+                period_halve_button,
+                width=width,
+                sizing_mode="scale_width",
+            ),
+            phase_selection,
+            period_selection,
+            width=width,
+        )
+    else:
```
Comment (guynir42) [inline skyportal/plot.py:1145]: I would make all these cases where it needs to be one of "flux", "mag" or "period" explicit `elif` for all three options, and add an `else` with a `ValueError`. 

Even though these are internal functions and it is very unlikely that the wrong key would ever be used, it makes it clear that the last `elif` is for "period" and not something generic.

**Review by guynir42 — APPROVED** [review]
Looks great! Minor suggestions below.

File: skyportal/plot.py (hunk 2 above)
Comment (jadalilleboe) [reply skyportal/plot.py:1146]: Good point, updated that.

File: skyportal/plot.py [hunk 3]
```diff
@@ -808,7 +808,7 @@ def make_legend_items_and_detection_lines(
     data : pandas DataFrame object
         Photometry data (ungrouped)
     obsind : pandas Series object
-        ???
+        Index of points that have been positively detected (observed), i.e., points that are not upper limits.
```
Comment (guynir42) [inline skyportal/plot.py:811]:
```suggestion
        Index of points that have been positively detected (observed), 
        i.e., points that are not upper limits.
```

File: skyportal/plot.py [hunk 4]
```diff
@@ -1243,7 +1246,7 @@ def make_photometry_panel(panel_name, device, width, user, data, obj_id, spectra
         The source/object's spectra
     Returns
     -------
-    bokeh Panel object
+    bokeh Panel object or None if the panel should not be added to the plot (i.e. no period plot)
```
Comment (guynir42) [inline skyportal/plot.py:1249]:
```suggestion
    bokeh Panel object or None if the panel should 
    not be added to the plot (i.e. no period plot)
```

**Review by guynir42 — APPROVED** [review]

**Comment by guynir42** [comment]
I can't push to this repo. @jadalilleboe try to pull down the changes, then make any change to the file and commit. The pre-commit hooks should fix the white space.

**Comment by guynir42** [comment]
@profjsb the tests are now :heavy_check_mark:

**Review by profjsb — APPROVED** [review]
LGTM (I did not review the code but did try locally).

**Pull request merged**